mixnode/src/commands/upgrade.rs

```diff
@@ -244,15 +245,87 @@ fn patch_010_upgrade(
 
     let upgraded_config = config.with_custom_version(to_version.to_string().as_ref());
 
-    upgraded_config.save_to_file(None).unwrap_or_else(|err| {
-        eprintln!("failed to overwrite config file! - {:?}", err);
-        print_failed_upgrade(&config_version, &to_version);
-        process::exit(1);
-    });
+    upgraded_config.save_to_file(None).map_err(|err| {
+        (
+            to_version.clone(),
+            format!("failed to overwrite config file! - {:?}", err),
+        )
+    })?;
 
     print_successful_upgrade(config_version, to_version);
 
-    upgraded_config
+    Ok(upgraded_config)
+}
+
+fn patch_0_10_2_upgrade(
+    config: Config,
+    _matches: &ArgMatches,
+    config_version: &Version,
+    package_version: &Version,
+) -> Result<Config, (Version, String)> {
+    #[derive(Deserialize)]
+    struct OldNodeDescription {
+        name: String,
+        description: String,
+        link: String,
+    }
+    let to_version = package_version;
+    let id = config.get_id();
+    let config_path = Config::default_config_directory(&id);
+
+    print_start_upgrade(&config_version, &to_version);
+
+    let upgraded_config = config.with_custom_version(to_version.to_string().as_ref());
+
+    upgraded_config.save_to_file(None).map_err(|err| {
+        (
+            to_version.clone(),
+            format!("failed to overwrite config file! - {:?}", err),
+        )
+    })?;
+
+    let description_file_path: PathBuf = [config_path.to_str().unwrap(), DESCRIPTION_FILE]
```

Comment:
You can just use a `join()` on Path/PathBuf rather than create an array and convert the iterator into a PathBuf

Reply:
According to the [man page](https://doc.rust-lang.org/std/path/struct.PathBuf.html), `join` (as well as `push`) should be used for dynamic situations. They give this iterator to `PathBuf` as example, which fits as well in our situation.

Reply:
Fair enough - I haven't realised that